src/components/Card.js

```diff
@@ -1,58 +1,122 @@
         this._name = data.name;
         this._link = data.link;
+        this._cardOwnerId = data.owner._id;
+        this._cardId = data._id;
+        this._cardLikes = data.likes;
 
         // Find the card selector
         this._cardSelector = cardSelector;
 
-        // Handle the image-click function
+        // Find the user id
+        this._userId = userId;
+
+        // Functions
         this._handleImageClick = handleImageClick;
+        this._handleDeleteClick = handleDeleteClick;
+        this._handleLikeClick = handleLikeClick;
     }
 
+    // Set the card's event listeners
     _setEventListeners() {
 
         // Add event listner for like button
         this._likeButton = this._element.querySelector('.card__like-button');
-        this._likeButton.addEventListener('click', this._handleLikeIcon);
-
-        // Add event listener for the delete button
-        const deleteButton = this._element.querySelector('.card__delete-button');
-        deleteButton.addEventListener('click', () => this._handleDeleteCard());
+        this._likeButton.addEventListener('click', this._handleLikeClick);
 
         // Add event listener for image
         this._imageWindow = this._element.querySelector('.card__image');
         this._imageWindow.addEventListener('click', () => this._handleImageClick({link: this._link, name: this._name}));
     }
 
-    _handleLikeIcon = () => {
-        // Add active class to card's like button
-        this._likeButton.classList.toggle('card__like-button_active');
+    // Check if the card is liked by the user
+    isLiked() {
+        return this._cardLikes.some((cardLike) => {
+            return cardLike._id === this._userId;
+        })
+    }
+
+    // Update the card's number of likes
+    updateLikeCount(likes) {
+        this._cardLikes = likes;
+        return this._likeCounter.textContent = likes.length;
+    }
+
+    // Display the number of likes
+    displayLikeCount() {
+        return this._likeCounter.textContent = this._cardLikes.length;
+    }
+
+    // Display the like icon based on its status
+    displayLikeIcon() {
+        if (this.isLiked()) {
+            this._likeButton.classList.add('card__like-button_active');
+        } else {
+            this._likeButton.classList.remove('card__like-button_active');
+        }
+    }
+
+    // Activate the like icon
+    addLikeIcon() {
+        this._likeButton.classList.add('card__like-button_active');
+    }
+
+    // Deactivate the like icon
+    removeLikeIcon() {
+        this._likeButton.classList.remove('card__like-button_active');
     }
 
-    _handleDeleteCard() {
+    // Display bin icon on cards created by the user
+    _handleDeleteButton() {
+
+        // Check if the user id matches the card owner's id
+        if (this._userId !== this._cardOwnerId) {
+
+            // If it does, remove the delete button
+            this._deleteButton.remove();
+        }
+    }
+
+    // Handle the deletion of cards
+    handleDeleteCard() {
```

Comment:
Function name should describe what the function does. This is not a listener anymore, so it should be named something like `deleteCard`

Reply:
Done! Thank you.